nncf/experimental/onnx/datasets/imagenet_dataset.py

```diff
@@ -47,18 +47,32 @@ def __len__(self) -> int:
         return len(self.dataset)
 
 
-def infer_input_shape(model: ModelProto) -> Tuple[Optional[int], int, int, int]:
-    assert len(model.graph.input) == 1
-
+def infer_input_shape(model: ModelProto, main_shape: List[int] = None) -> Tuple[List[int], List[str]]:
+    input_shape = None
+    input_keys = None
     for _input in model.graph.input:
-        dim = _input.type.tensor_type.shape.dim
-        input_shape = [MessageToDict(d).get("dimValue") for d in dim]
-
-    input_shape = [int(dim) if dim is not None else dim for dim in input_shape]
-
-    assert len(input_shape) == 4
-
-    return input_shape
+        if len(_input.type.tensor_type.shape.dim) == 4:
+            dim = _input.type.tensor_type.shape.dim
+            _input_shape = [int(MessageToDict(d).get("dimValue")) for d in dim]
+            if main_shape is not None:
+                if main_shape == _input_shape:
+                    input_shape = main_shape
+                    input_keys = _input.name
+                    break
+            else:
+                if input_shape is None and input_keys is None:
+                    input_shape = _input_shape
+                    input_keys = _input.name
+                else:
+                    if input_shape[2:] < _input_shape[2:]:
+                        # If _input_shape > input_shape, input_shape is not a real input size, 
+                        # so replace it with _input_shape.
+                        input_shape = _input_shape
+                        input_keys = _input.name
```

Comment:
First of all, I think it's more natural to ask the user the name of the input tensor rather than the shape of the input tensor.
Secondly, we need to avoid nested if-else statements for the better readability.
```
assert len(model.graph.input) > 0

if main_shape is given:
     for loop
else:
     input_shape = model.graph.input[0].name
     input_keys = model.graph.input[0].name
     for loop
```

Reply:
I fixed to ask user either the input tensor's name or shape and to find the proper tensor shape and name. Also, I removed unnecessary nested if-else.